tests/unit/src/test/scala/tests/CodeActionLspSuite.scala

```diff
@@ -0,0 +1,57 @@
+package tests
+
+import scala.meta.internal.metals.QuickFix.ImportMissingSymbol
+import scala.meta.internal.metals.MetalsEnrichments._
+
+object CodeActionLspSuite extends BaseLspSuite("codeAction") {
+
+  check(
+    "auto-import",
+    """|package a
+       |
+       |object A {
+       |  val f = Fut@@ure.successful(2)
+       |}
+       |""".stripMargin,
+    s"""|${ImportMissingSymbol.label("Future", "scala.concurrent")}
+        |${ImportMissingSymbol.label("Future", "java.util.concurrent")}
+        |""".stripMargin,
+    """|package a
+       |
+       |import scala.concurrent.Future
+       |
+       |object A {
+       |  val f = Future.successful(2)
+       |}
+       |""".stripMargin
+  )
```

Comment:
@tgodzik this is my proposal. You have three main elements:

- the input text (using `@@` as the cursor position)
- the expected list of code actions (I'm checking the `title` property of code actions)
- the expected text after applying one of the code actions (I'm selecting the first one by default, but it's configurable)

This can be expanded to support other common test cases, but I think it's a decent start

Reply:
That's exactly what I was thinking about. Looks awesome!